crates/nu-command/src/filters/flatten.rs

```diff
@@ -116,12 +117,17 @@ fn flatten(
     let metadata = input.metadata();
     let flatten_all = call.has_flag("all");
 
-    input
-        .flat_map(
-            move |item| flat_value(&columns, &item, tag, flatten_all),
-            engine_state.ctrlc.clone(),
-        )
-        .map(|x| x.set_metadata(metadata))
+    let input_iter = input.into_interruptible_iter(engine_state.ctrlc.clone());
+    let mut iter_val_vec = Vec::new();
+    for item in input_iter {
+        match flat_value(&columns, &item, tag, flatten_all) {
```

Comment:
This looks like it switches the lazy stream into not being lazy. As much as possible, the pipelines should work like normal iterators so values are computed only when needed.

Is there a different way we can solve it that doesn't compute the whole stream first?

Reply:
I think I'm seeing the problem here. When the input is being processed, it gets streamed out with `PipelineData` already being returned. However, if a ShellError is encountered, there is no way for `run` to return an `Err` value. 

I'm not sure if there's an easy fix to this, but I'll keep trying.